### PR TITLE
menu cleanup, fix crash when switching tabs with mouse wheel

### PR DESCRIPTION
**Pull request by rfomin**
* Bring back alternative default description.

* Use designators in structs initialization.

### CODE REVIEW

File: src/mn_internal.h [hunk 1]
```diff
@@ -197,6 +198,7 @@ typedef struct setup_menu_s
     void (*action)(void); // killough 10/98: function to call after changing
     mrect_t rect;
     int lines;            // number of lines for rect, always > 0
+    const char *desc;     // alternative default description
```
Comment (fabiangreffrath) [inline src/mn_internal.h:201]: Also `altdesc`?

**Review by fabiangreffrath — APPROVED** [review]
I like this style.

File: src/mn_setup.c [hunk 2]
```diff
@@ -1132,12 +1132,9 @@ static void DrawInstructions()
 
     const char *s = "";
 
-    if (flags & S_FUNC)
+    if (flags & S_ALTDESC)
```
Comment (fabiangreffrath) [inline src/mn_setup.c:1135]: Sufficient to check for `item->desc` here? I mean, do we need the flag actually?

Comment (rfomin) [reply src/mn_setup.c]: Right, I found real bug here: https://github.com/fabiangreffrath/woof/pull/1861/commits/1692cb580ea45a0cd0caefdec6cb2f2d72e1a196

**Pull request merged**